_RFCs/0007-deprecating-packages.md

```diff
@@ -0,0 +1,65 @@
+- Change Name: deprecate_unsupported_software_plans
+- Start Date: 2018-05-07
+- RFC PR: (leave this empty)
+- Core-Plans Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+When a piece of software that is packaged with a Habitat plan is no longer supported by the manufacturer of that software, we need a process to deprecate that plan.
+
+Deprecating a plan does NOT involve removing it from builder. Rather, we must indicate to users that the plan will no longer be updated by the Habitat team except in cases of security vulnerabilities.
+
+This RFC proposes a process to:
+* Deprecate a plan
+* Communicate that deprecation notice to all users
+* Establish a process for handling emergency updates - like serious regression bugs
+
+# Motivation
+[motivation]: #motivation
+
+We need a way to indicate that a plan is deprecated without removing the plan itself from Builder (as this could potentially cause breakages for anyone who is still dependent on it), and to redirect users to updated plans that are still supported by the manufacturer.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+We currently have plans that package software which the author/manufacturer has declared EOL or otherwise no longer updated. This includes, but may not be limited to:
+* [core/ruby22](https://github.com/habitat-sh/core-plans/tree/master/ruby22) (The Ruby team has indicated [here](https://www.ruby-lang.org/en/news/2018/03/28/ruby-2-2-10-released/) that Ruby 2.2.x security maintenance ended in March 2018. Additionally, we discovered during the latest base plans refresh that Ruby 2.2.x does not work with newer versions of glibc and gcc. While we can pin it to older versions of glibc and gcc, this will introduce potential dependency issues when used with software that was created with newer versions of those base packages)
+* [core/jre9](https://github.com/habitat-sh/core-plans/tree/master/jre9)(Oracle has indicated [here](http://www.oracle.com/technetwork/java/javase/downloads/jre9-downloads-3848532.html) that Java SE 9 has reached end of support and users should upgrade to Java SE 10. Additionally, Oracle's download links to JRE 9 and JDK 9 have been removed. It should be noted that Oracle DOES still support JRE 8 and JDK 8.)
+* [core/jdk9](https://github.com/habitat-sh/core-plans/blob/master/jdk9/plan.sh)(see explanation for core/jre9)
+
+When we become aware that a piece of software we package in core plans has been EOL'd or the author/manufacturer otherwise stops supporting it, we will:
+* Open an RFC explaining why the plan must be deprecated
+* When that RFC is approved (though the normal RFC approval process), we must
+1) Delist the package in builder by setting it to "private" (that will make it invisible to anyone outside of the core org, as mentioned [in this issue](https://github.com/habitat-sh/builder/issues/18)
```

Comment:
I think we may be over-leveraging the intent of "private".  Maybe we should consider having a "deprecated" option in the API/UI as well.